src/rust.rs

```diff
@@ -1635,3 +1635,101 @@ macro_rules! capture_stack {
         let $name = $crate::rust::CapturedJSStack::new($cx, __obj, None);
     }
 }
+
+macro_rules! wrap {
+    // The invocation of @inner has the following form:
+    // @inner (input args) <> (accumulator) <> unparsed tokens
+    // when `unparsed tokens == \eps`, accumulator contains the final result
+
+    (@inner $saved:tt <> ($($acc:expr,)*) <> $arg:ident: Handle<$gentype:ty>, $($rest:tt)*) => {
+        wrap!(@inner $saved <> ($($acc,)* $arg.into(),) <> $($rest)*);
+    };
+    (@inner $saved:tt <> ($($acc:expr,)*) <> $arg:ident: MutableHandle<$gentype:ty>, $($rest:tt)*) => {
+        wrap!(@inner $saved <> ($($acc,)* $arg.into(),) <> $($rest)*);
+    };
+    (@inner $saved:tt <> ($($acc:expr,)*) <> $arg:ident: Handle, $($rest:tt)*) => {
+        wrap!(@inner $saved <> ($($acc,)* $arg.into(),) <> $($rest)*);
+    };
+    (@inner $saved:tt <> ($($acc:expr,)*) <> $arg:ident: HandleValue, $($rest:tt)*) => {
+        wrap!(@inner $saved <> ($($acc,)* $arg.into(),) <> $($rest)*);
+    };
+    (@inner $saved:tt <> ($($acc:expr,)*) <> $arg:ident: HandleObject, $($rest:tt)*) => {
+        wrap!(@inner $saved <> ($($acc,)* $arg.into(),) <> $($rest)*);
+    };
+    (@inner $saved:tt <> ($($acc:expr,)*) <> $arg:ident: HandleId, $($rest:tt)*) => {
+        wrap!(@inner $saved <> ($($acc,)* $arg.into(),) <> $($rest)*);
+    };
+    (@inner $saved:tt <> ($($acc:expr,)*) <> $arg:ident: MutableHandle, $($rest:tt)*) => {
+        wrap!(@inner $saved <> ($($acc,)* $arg.into(),) <> $($rest)*);
+    };
+    (@inner $saved:tt <> ($($acc:expr,)*) <> $arg:ident: MutableHandleObject, $($rest:tt)*) => {
+        wrap!(@inner $saved <> ($($acc,)* $arg.into(),) <> $($rest)*);
+    };
+    (@inner $saved:tt <> ($($acc:expr,)*) <> $arg:ident: MutableHandleValue, $($rest:tt)*) => {
+        wrap!(@inner $saved <> ($($acc,)* $arg.into(),) <> $($rest)*);
+    };
+    (@inner $saved:tt <> ($($acc:expr,)*) <> $arg:ident: $type:ty, $($rest:tt)*) => {
+        wrap!(@inner $saved <> ($($acc,)* $arg,) <> $($rest)*);
+    };
+    (@inner ($func_name:ident ($($args:tt)*) -> $outtype:ty) <> ($($argexprs:expr,)*) <> ) => {
+        pub unsafe fn $func_name($($args)*) -> $outtype {
+            jsapi::$func_name($($argexprs),*)
+        }
+    };
+    (pub fn $func_name:ident($($args:tt)*) -> $outtype:ty) => {
+        wrap!(@inner ($func_name ($($args)*) -> $outtype) <> () <> $($args)* ,);
+    };
+    (pub fn $func_name:ident($($args:tt)*)) => {
+        wrap!(pub fn $func_name($($args)*) -> ());
+    }
+}
+pub mod wrappers {
```

Comment:
Let's add documentation like:
```
Wrappers for JSAPI methods that accept FFI Handle and MutableHandle arguments. The wrapped methods are identical except that they accept Handle and MutableHandle arguments that include lifetimes instead.
```

Reply:
You mean:
> Wrappers for JSAPI methods that accept lifetimed Handle and MutableHandle arguments.
?

Reply:
My comment scrolls horizontally, so keep reading.